hugr-core/src/extension/op_def.rs

```diff
@@ -147,21 +139,22 @@ impl CustomValidator {
     }
 }
 
-/// The two ways in which an OpDef may compute the Signature of each operation node.
-#[derive(serde::Deserialize, serde::Serialize)]
+/// The ways in which an OpDef may compute the Signature of each operation node.
 pub enum SignatureFunc {
-    // Note: except for serialization, we could have type schemes just implement the same
-    // CustomSignatureFunc trait too, and replace this enum with Box<dyn CustomSignatureFunc>.
-    // However instead we treat all CustomFunc's as non-serializable.
-    /// A PolyFuncType (polymorphic function type), with optional custom
-    /// validation for provided type arguments,
-    #[serde(rename = "signature")]
-    PolyFuncType(CustomValidator),
-    #[serde(skip)]
+    /// An explicit polymorphic function type.
+    PolyFuncType(PolyFuncTypeRV),
+    /// A polymorphic function type with a custom binary for validating type arguments.
```

Comment:
maybe we could slightly strengthen that this is mostly the same as the previous case but PLUS a custom binary